src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -460,50 +481,60 @@ private static boolean isInitializationSequence(
             DetailAST variableUsageAst, String variableName) {
         boolean result = true;
         boolean isUsedVariableDeclarationFound = false;
+        boolean isFirstMethodCallToBeExamined = true;
+        boolean isInitialMethodCalledWithInstance = false;
+        boolean anotherVariableDeclarationFound = false;
         DetailAST currentSiblingAst = variableUsageAst;
-        String initInstanceName = "";
+        String initialMethodOrInstanceName = "";
 
         while (result
                 && !isUsedVariableDeclarationFound
                 && currentSiblingAst != null) {
```

Comment:
@romani Sonar can be cheated by extracting this condition.
```suggestion
        while (cannotThinkOfAResonableName(result, isUsedVariableDeclarationFound, currentSiblingAst)) {
```

Reply:
Please apply cheating update

Reply:
Difficult to name this condition, how about:
```java

    while (!isInitializationSequenceFound(result, isUsedVariableDeclarationFound, currentSiblingAst)) {

....

    private static boolean isInitializationSequenceFound(boolean result, boolean isUsedVariableDeclarationFound, DetailAST currentSiblingAst) {
        return !result
                || isUsedVariableDeclarationFound
                || currentSiblingAst == null;
    }
```

Reply:
@Vyom-Yadav won't some check fail on boolean method parameters?

Reply:
> @Vyom-Yadav won't some check fail on boolean method parameters?

`mvn clean verify` passed on local, don't know about sonar and other lints.

Reply:
lets not overhack code, keep it as is